.github/workflows/pytest.yml

```diff
@@ -1,41 +1,50 @@
 # This workflow will install Python dependencies, run tests and lint with a variety of Python versions
 # For more information see: https://help.github.com/actions/language-and-framework-guides/using-python-with-github-actions
 
-name: Python package
+name: Pyfive Test
 
 on:
   push:
-    branches: [ master ]
+    branches: [ master, modernize_package ]  # leave only "master" after PR69 is merged
   pull_request:
     branches: [ master ]
+  schedule:
+    - cron: '0 0 * * *'  # nightly
 
 jobs:
   build:
 
+    name: Test Python ${{ matrix.python-version }}
     runs-on: ubuntu-latest
     strategy:
       fail-fast: false
       matrix:
-        python-version: ["3.8", "3.9", "3.10", "3.11", "3.12", "3.13"]
+        python-version: ["3.10", "3.11", "3.12", "3.13"]
 
     steps:
-    - uses: actions/checkout@v3
-    - name: Set up Python ${{ matrix.python-version }}
-      uses: actions/setup-python@v3
-      with:
-        python-version: ${{ matrix.python-version }}
-    - name: Install dependencies
-      run: |
-        python -m pip install --upgrade pip
-        python -m pip install flake8 pytest
-        python -m pip install -e .
-        if [ -f requirements.txt ]; then pip install -r requirements.txt; fi
-    - name: Lint with flake8
-      run: |
-        # stop the build if there are Python syntax errors or undefined names
-        flake8 . --count --select=E9,F63,F7,F82 --show-source --statistics
-        # exit-zero treats all errors as warnings. The GitHub editor is 127 chars wide
-        flake8 . --count --exit-zero --max-complexity=10 --max-line-length=127 --statistics
-    - name: Test with pytest
-      run: |
-        pytest
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      - uses: conda-incubator/setup-miniconda@v3
+        with:
+          activate-environment: pyfive
+          environment-file: environment.yml
+          python-version: ${{ matrix.python-version }}
+          miniforge-version: "latest"
+          use-mamba: true
+      - name: Install Pyfive Test Mode
+        run: |
+          pip install -e .[test]
+      - name: Lint with flake8
```

Comment:
Would it make sense to split out linting in a separate job? For the packages I maintain I find it useful to see test breakages even if the linting fails.

Reply:
that's entirely up to you, folks! We stop and fail the workflow if eg `pre-commit` fails, but I guess that depends how much emphasis one project puts on linting. Eg here we can put a `fail-fast: false` clause so the workflow still fails but it runs all the steps without halting at the first failed step?

Reply:
I just noticed that the `strategy` already has a `fail-fast: false` so this would mean workflow runs all the steps and marks the wkflow as failed if one step failed, so we good here I reckon :+1:

Reply:
Great! I must have overlooked the `fail-fast` setting.